include/deal.II/base/bounding_box.h

```diff
@@ -0,0 +1,108 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 1998 - 2016 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_base_bounding_box_h
+#define dealii_base_bounding_box_h
+
+
+#include <deal.II/base/config.h>
+#include <deal.II/base/exceptions.h>
+#include <deal.II/base/point.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+/**
+ * A class that represents a bounding box in a space with arbitrary dimension
+ * <tt>spacedim</tt>.
+ *
+ * Objects of this class are used to represent bounding boxes. They are,
+ * among other uses, useful in parallel distributed meshes to give a general
+ * description the owners of each portion of the mesh.
+ *
+ * Boundig boxes are represented by two vertices and, geometrically, are
+ * represented by the following:
+ * 1 d: a segment (represented by its vertices)
+ * 2 d: a rectangle (represented by two vertices V opposed to the center)
+ * V--------.
+ * |        |
+ * .--------V
+ * 3 d: a cuboid (represented by two vertices V opposed to the center
+ *               and not owned by the same face)
+ *   V------.
+ *  /      /|
+ * .------. |
+ * |      | /
+ * |      |/
+ * .------V
+ *
+ * Notice the sides are always parallel to the respective axis.
+ *
+ * @author Giovanni Alzetta, 2017
+ */
+template <int spacedim, typename Number=double>
+class BoundingBox
+{
+public:
+  /**
+   * Standard constructor. Creates an object that corresponds to a null box,
+   * i.e. an degenerate box with both points being the origin.
+   */
+  BoundingBox ();
+
+  /**
+   * Standard constructor for non-empty boxes: it uses a pair of points
+   * which describe the box: one for the bottom and one for the top
+   * corner.
+   */
+  BoundingBox (const std::pair<Point<spacedim,Number>,Point<spacedim,Number>> &boundary_points); //Add two points constructor
+
+  /**
+   * Returns the boundary_points
+   */
+  std::pair<Point<spacedim,Number>,Point<spacedim,Number>> &get_boundary_points ();
```

Comment:
either make this
```
const std::pair<Point<spacedim,Number>,Point<spacedim,Number>> &get_boundary_points () const;
```
or add the `const` version.

Reply:
on a second though, i think there should be no write access to points. If they are to be modified, then we need a separate method `set()` or something like that where we can make sure that the provided input makes sense.

Reply:
I agree: for what I need there's no need to modify them so I created only the const version.

Reply:
And thanks for the 200 comments, I didn't think the code would be so sloppy. I'll finish fixing and hopefully this part shall be ok